volatility3/framework/plugins/windows/netscan.py

```diff
@@ -142,76 +142,94 @@ def determine_tcpip_version(cls, context: interfaces.context.ContextInterface, l
-                (6, 1, 7600): "netscan-win7-x64",
-                (6, 1, 7601): "netscan-win7-x64",
-                (6, 1, 8400): "netscan-win7-x64",
-                (6, 2, 9200): "netscan-win8-x64",
-                (6, 3, 9600): "netscan-win81-x64",
-                (10, 0, 10240): "netscan-win10-x64",
-                (10, 0, 10586): "netscan-win10-x64",
-                (10, 0, 14393): "netscan-win10-x64",
-                (10, 0, 15063): "netscan-win10-15063-x64",
-                (10, 0, 16299): "netscan-win10-16299-x64",
-                (10, 0, 17134): "netscan-win10-17134-x64",
-                (10, 0, 17763): "netscan-win10-17763-x64",
-                (10, 0, 18362): "netscan-win10-18362-x64",
-                (10, 0, 18363): "netscan-win10-18363-x64",
-                (10, 0, 19041): "netscan-win10-19041-x64"
+                (6, 0, 6000, 0): "netscan-vista-x64",
+                (6, 0, 6001, 0): "netscan-vista-sp12-x64",
+                (6, 0, 6002, 0): "netscan-vista-sp12-x64",
+                (6, 0, 6003, 0): "netscan-vista-sp12-x64",
+                (6, 1, 7600, 0): "netscan-win7-x64",
+                (6, 1, 7601, 0): "netscan-win7-x64",
+                (6, 1, 8400, 0): "netscan-win7-x64",
+                (6, 2, 9200, 0): "netscan-win8-x64",
+                (6, 3, 9600, 0): "netscan-win81-x64",
+                (6, 3, 9600, 19935): "netscan-win81-19935-x64",
+                (10, 0, 10240, 0): "netscan-win10-x64",
+                (10, 0, 10586, 0): "netscan-win10-x64",
+                (10, 0, 14393, 0): "netscan-win10-x64",
+                (10, 0, 15063, 0): "netscan-win10-15063-x64",
+                (10, 0, 16299, 0): "netscan-win10-16299-x64",
+                (10, 0, 17134, 0): "netscan-win10-17134-x64",
+                (10, 0, 17763, 0): "netscan-win10-17763-x64",
+                (10, 0, 18362, 0): "netscan-win10-18362-x64",
+                (10, 0, 18363, 0): "netscan-win10-18363-x64",
+                (10, 0, 19041, 0): "netscan-win10-19041-x64"
             }
 
-        # special use case: Win10_18363 is not recognized by windows.info as 18363
+        # we do not need to check for tcpip's specific FileVersion in every case
+        tcpip_mod_version = 0 # keep it 0 as a default
+
+        # special use cases
+
+        # Win10_18363 is not recognized by windows.info as 18363
         # because all kernel file headers and debug structures report 18363 as
         # "10.0.18362.1198" with the last part being incremented. However, we can use
         # os_distinguisher to differentiate between 18362 and 18363
         if vers_minor_version == 18362 and is_18363_or_later:
             vollog.debug("Detected 18363 data structures: working with 18363 symbol table.")
             vers_minor_version = 18363
 
+        # Win 8 SP 1 also may have different structures based on specific tcpip.sys version
+        if (nt_major_version, nt_minor_version, vers_minor_version) == (6, 3, 9600):
```

Comment:
Seems a little specific?  Is there way of checking if there tcpip.sys options?  Something like:

```
if [ a,b,c,d for x in version_dict if (a,b,c) == (nt_major_version, nt_minor_version, vers_minor_version) and d != 0]:
```
Ideally I'd prefer longer variable names since but it depends how clean that would look (and for this comment, short is fine).

Reply:
IMO we only need the tcpip_mod_version number for certain Windows versions (such as 6.3.9600 / 8.1). I don't think we need to check the FileVersion header in all other cases where we don't have ambiguity. 

I'd suggest adding more options to this condition if we discover more cases where we need to consider this header. Something like this, maybe.
```python
if (nt_major_version, nt_minor_version, vers_minor_version) in [(6, 3, 9600), (10, 0, 19042), ...]:
    ...
```

Reply:
Errrr, ok, but it feels like adding a new fix requires changing two places, I'd prefer people only need to add to one place and the rest of the code just does the right thing?   Also that list could one day get long enough to make the code very ugly, and I'd prefer to code for the future than for the right now.  I'm not gonna push it if you really object, but I'm gonna need one more "I really think this is better" before I fold...  5;P

Reply:
You're right. I've added it like this - https://github.com/volatilityfoundation/volatility3/pull/476/commits/10c1808e22f68921f7538659f34f3428d84bc389 - I hope the short var names are OK for now. I think at this point it appears necessary to create some more extensive docs about netscan & netstat and their peculiarities anyway...

Reply:
Yeah, in this instance they're ok.  I believe we now cover the majority of situations, which is good, so we're narrowing the bad results down further and further.  5:)